### PR TITLE
Allow rename for symbols that are both defined in user code and generated

### PR DESCRIPTION
**Pull request by ryzngard**
Fixes [AB#1617866](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1617866)

### CODE REVIEW

File: src/EditorFeatures/Core/InlineRename/AbstractEditorInlineRenameService.cs [hunk 1]
```diff
@@ -139,8 +139,9 @@ public async Task<IInlineRenameInfo> GetRenameInfoAsync(Document document, int p
 
                     if (sourceDocument is SourceGeneratedDocument)
                     {
-                        // The file is generated so we can't go editing it (for now)
-                        return new FailureInlineRenameInfo(EditorFeaturesResources.You_cannot_rename_this_element);
+                        // The file is generated so doesn't count towards valid spans 
+                        // we can edit.
+                        continue;
```
Comment (ryzngard) [inline src/EditorFeatures/Core/InlineRename/AbstractEditorInlineRenameService.cs:144]: I didn't see any tests with this so I think I'll need to add one. The test is easy with .NET 7 and `GeneratedRegex`. Can we use .NET 7 for tests though?

Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core/InlineRename/AbstractEditorInlineRenameService.cs:144]: i don't have a problem with this, but what does this mean wrt to the user experience.  e.g. if i'm in the source-generated doc, can i start an inline rename session, and will it allow both editing of that span, while we're also running gneerators and regen'ing the file?  it's kinda unclear what will happen.

Comment (ryzngard) [reply src/EditorFeatures/Core/InlineRename/AbstractEditorInlineRenameService.cs:144]: I'll double check, but I believe it should be supported. Need to make sure the user experience isn't odd though. Good call out

Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core/InlineRename/AbstractEditorInlineRenameService.cs:144]: Def worth checking. Especially if the SG file is read only. Thanks!

Comment (ryzngard) [reply src/EditorFeatures/Core/InlineRename/AbstractEditorInlineRenameService.cs:144]: If invoked on a readonly SG file, it gives an alert that the file is readonly and can't perform rename. Are there cases where SG are not readonly? I wasn't able to find one.

**Pull request closed without merging**